From fdeb3a3d71cbbe02baf07af9c82aeeea7dc4325f Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:37 -0400
Subject: [ethernet] be2net: define BE_MAX_MTU

Message-id: <1410192881-30702-75-git-send-email-ivecera@redhat.com>
Patchwork-id: 92453
O-Subject: [RHEL7.1 PATCH 74/78] be2net: define BE_MAX_MTU
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 0d3f5ccec907bcfb4f5d4941d762dc2f89b2d24d
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Tue Sep 2 09:56:53 2014 +0530

    be2net: define BE_MAX_MTU

    This patch defines a new macro BE_MAX_MTU to make the code in be_change_mtu()
    more readable.

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be.h b/drivers/net/ethernet/emulex/benet/be.h
index e5ba83e..07653da 100644
--- a/drivers/net/ethernet/emulex/benet/be.h
+++ b/drivers/net/ethernet/emulex/benet/be.h
@@ -86,6 +86,8 @@ static inline char *nic_name(struct pci_dev *pdev)
 
 #define BE_MAX_JUMBO_FRAME_SIZE 9018
 #define BE_MIN_MTU  256
+#define BE_MAX_MTU              (BE_MAX_JUMBO_FRAME_SIZE - \
+     (ETH_HLEN + ETH_FCS_LEN))
 
 #define BE_NUM_VLANS_SUPPORTED 64
 #define BE_MAX_EQD  128u
diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 927410d..9e52d05 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -1073,15 +1073,15 @@ static netdev_tx_t be_xmit(struct sk_buff *skb, struct net_device *netdev)
 static int be_change_mtu(struct net_device *netdev, int new_mtu)
 {
  struct be_adapter *adapter = netdev_priv(netdev);
- if (new_mtu < BE_MIN_MTU ||
-     new_mtu > (BE_MAX_JUMBO_FRAME_SIZE - (ETH_HLEN + ETH_FCS_LEN))) {
-  dev_info(&adapter->pdev->dev,
-    "MTU must be between %d and %d bytes\n",
-    BE_MIN_MTU,
-    (BE_MAX_JUMBO_FRAME_SIZE - (ETH_HLEN + ETH_FCS_LEN)));
+ struct device *dev = &adapter->pdev->dev;
+
+ if (new_mtu < BE_MIN_MTU || new_mtu > BE_MAX_MTU) {
+  dev_info(dev, "MTU must be between %d and %d bytes\n",
+    BE_MIN_MTU, BE_MAX_MTU);
   return -EINVAL;
  }
- dev_info(&adapter->pdev->dev, "MTU changed from %d to %d bytes\n",
+
+ dev_info(dev, "MTU changed from %d to %d bytes\n",
    netdev->mtu, new_mtu);
  netdev->mtu = new_mtu;
  return 0;
-- 
1.7.1